packages/explat/src/assignment.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { stringify } from 'qs';
+
+const EXPLAT_VERSION = '0.1.0';
+
+export const fetchExperimentAssignment = async ( {
+	experimentName,
+	anonId,
+}: {
+	experimentName: string;
+	anonId: string | null;
+} ): Promise< unknown > => {
+	if ( ! window.wcTracks.isEnabled ) {
+		throw new Error(
```

Comment:
Hmm, there is actually an interesting point to be made here about experiment loading behaviour after 

I may have been wrong about this, [your suggestion here](https://github.com/woocommerce/woocommerce-admin/pull/6669#discussion_r608174109) might be along the right tracks:

> returning `{"variations":{},"ttl":60}`

Particularly, I think we want to return `0` ttl here as once experiments are enabled we immediately want to be able to load them. Currently in the client throws an error for a `0` ttl but if we think it is a good idea I can remove that constraint.

Reply:
> Particularly, I think we want to return 0 ttl here as once experiments are enabled we immediately want to be able to load them. Currently in the client throws an error for a 0 ttl but if we think it is a good idea I can remove that constraint.

Just to elaborate on this, the `ttl` is used to determine when to refetch the data, it is `60` in dev environments `3600` in live environments, so _if an experiment attempts to be loaded before `wctracks` is loaded_, it will take 1 hour before an experiment can apply to that user.

Reply:
>we immediately want to be able to load them

In my testing, once a user opts in and fires the callback in `wcTracks.enable()` to initialize the anon id, the assignments aren't fetched until the user lands on the page with the Experiment. Are you suggesting calling that function right away and caching the results?

Reply:
>if an experiment attempts to be loaded before `wctracks` is loaded, it will take 1 hour before an experiment can apply to that user.

I see, thanks for the context. If an fetchExperimentAssignment is called when the user isn't opted in an error is thrown and no `ttl` value is returned. Does that count as an attempt? If not, I think we are good in this respect.

https://github.com/woocommerce/woocommerce-admin/blob/ed6e18a042e1eb4c405bc9105d169c9a22b19256/packages/explat/src/assignment.ts#L15-L19

Reply:
> I see, thanks for the context. If an fetchExperimentAssignment is called when the user isn't opted in an error is thrown and no ttl value is returned. Does that count as an attempt? If not, I think we are good in this respect.

Ah my numbers are wrong actually: In the event of an error it doesn't fetch the `ttl` from the server but rather uses the default `60` seconds ttl.

This isn't horrible and will probably work well enough but might impact some edge cases at some point, this should be fine for the A/A experiment and something we can fix later.